ethereum/blockcreation/src/main/java/org/hyperledger/besu/ethereum/blockcreation/txselection/BlockTransactionSelector.java

```diff
@@ -161,7 +162,12 @@ public BlockTransactionSelector(
     blockWorldStateUpdater = worldState.updater();
     txWorldStateUpdater = blockWorldStateUpdater.updater();
     blockTxsSelectionMaxTime = miningConfiguration.getBlockTxsSelectionMaxTime();
-    balBuilder = new BlockAccessList.BlockAccessListBuilder();
+
+    maybeBlockAccessListBuilder =
+        protocolSpec
+            .getBlockAccessListFactory()
+            .filter(BlockAccessListFactory::isForkActivated)
```

Comment:
Won't this break CLI only activations of BAL?

Reply:
The CLI flag was not intended to and should not have any impact on block building.

As [the original PR](https://github.com/hyperledger/besu/pull/9012) description mentions:
> A new feature flag --Xapi-block-access-list-enabled enables:
> * The eth_getBlockAccessListByNumber RPC endpoint.
> * Caching BALs for the last cache-last-blocks imported blocks.
> * Inclusion of BALs in block simulation outputs and call result objects.